### ISSUE TITLE
Bug in SubmissionQueueEvent::prep_read_fixed

### ISSUE THREAD

**Issue by twissel** [withoutboats/iou#11]
Setting ``` self.sqe.flags |= SubmissionFlags::FIXED_FILE.bits();```  https://github.com/withoutboats/iou/blob/master/src/sqe.rs#L134 сausing ```Error: Os { code: 9, kind: Other, message: "Bad file descriptor" } ```
Code to reproduce: 
```
#[test]
fn test_read_fixed() -> io::Result<()> {
    let mut io_uring = iou::IoUring::new(32)?;

    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
    path.push("props");
    path.push("text.txt");
    let file = File::open(&path)?;
    let mut buf0 = [0; 4096];
    let mut buf1 = [0; 4096];
    let mut buffs = &[IoSlice::new(&buf0), IoSlice::new(&buf1)];
    let registrar = io_uring.registrar();
    registrar.register_buffers(buffs);
    unsafe {
        prep_fixed(&mut io_uring, &mut buf0, file.as_raw_fd())?;
    }

    let dirt = dirty_stack();

    let n = {
        let mut cq = io_uring.cq();
        let cqe = cq.wait_for_cqe()?;
        assert_eq!(cqe.user_data(), 0xDEADBEEF);
        cqe.result()? as usize
    };

    assert_eq!(&TEXT[..n], &buf0[..n]);
    drop(dirt);

    Ok(())
}

#[inline(never)]
unsafe fn prep_fixed(ring: &mut iou::IoUring, buf: &mut [u8], fd: RawFd) -> io::Result<()> {
    let mut sq = ring.sq();
    let mut sqe = sq.next_sqe().unwrap();
    sqe.prep_read_fixed(fd, buf, 0, 0);
    sqe.set_user_data(0xDEADBEEF);
    sq.submit()?;
    Ok(())
} 
```

**Comment by withoutboats** [comment]
Does this work if the code is patched so that the flag is not set?

My read of the io_uring PDF was that this flag was supposed to be set for fixed reads and writes, but perhaps I was mistaken.

**Comment by withoutboats** [comment]
I see, I've conflated *fixed files* and *fixed buffers*.

**Comment by twissel** [comment]
yeah

**Comment by twissel** [comment]
The same applies to write_fixed

**Comment by twissel** [comment]
If you don't mind, I will make a pull request to fix this issue(s)

**Comment by withoutboats** [comment]
Just fixed in master #13 :) Thanks for the report!
